semantic_conventions/resource/container.yaml

```diff
@@ -32,3 +32,25 @@ groups:
         brief: >
           Container image tag.
         examples: ['0.1']
+      - id: image.id
```

Comment:
I suggest to follow [OCI spec](https://github.com/opencontainers/image-spec/blob/0b40f0f367c396cc5a7d6a2e8c8842271d3d3844/descriptor.md#properties) terminology here.

I believe we should move OCI (image, artifact) properties out of the `container` namespace as they can be used for unrelated things (VM images, helm charts, and anything else). I propose adding new root `oci` namespace and declaring `oci.manfest.digest`  there

Reply:
Created https://github.com/open-telemetry/semantic-conventions/issues/48 to discuss as it's broader than scope of the current  PR

Reply:
Looking into this a bit more, docker has image `Id` property which seems to be different from image digest and is [undocumented(?)](https://stackoverflow.com/questions/56364643/whats-the-difference-between-a-docker-images-image-id-and-its-digest) thing. 

If there is a need to populate docker one, we should consider having `container.docker.image.id` and clearly explaining what it is and how to get it on docker. If we need a way to identify image regardless of environment, we should add something like `oci.image.digest`.

Reply:
Unsure how I feel about changing to an `oci` namespace. It's more generally correct but I also think it's less obvious for a user consuming those names. 

The [dockerstatsreceiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/dockerstatsreceiver/metadata.yaml) doesn't currently report image ID. But if it did, to me it feels strange that all of the other resource attributes coming out of that receiver are in the `container` namespace but then the image ID is under `oci`. 

Would be interested to hear other's thoughts.

Reply:
I definitely see the benefit. Do you suggest we wait until the resolution of #48 to add this in, or go with `oci` namespace straight away in this PR?

Reply:
I think it's ok to can start with `oci` attribute right away or, if we're not sure this attribute is necessary, we can just postpone adding it until we have a clear use case. I also was not sure if there was a desire to capture docker image Id specifically - then I'd propose a `container.docker` namespace for it. 
The #48 issue should be a blocker for container semconv stability, but is not strictly related to this PR and should not block it.

Reply:
We can also go with a `image.` namespace, since OCI seems to separate those into an entirely separated repo https://github.com/opencontainers/image-spec/blob/main/manifest.md. 

As for doing here or there, I feel the `image` discussion as a per-requisite. It can also be done as part of this PR. We introduce the new image yaml/attributes and then just reference them in `container.yaml`.

Reply:
the `oci` part feels important only because we have at least three things for image IDs:
- what docker uses - sha256 of something 
- what k8s uses  - a link to the container registry repo with digest    `"imageID": "registry.azurecr.io/namespace/service/dockerfile@sha256:bdeabd40c3a8a492eaf9e8e44d0ebbb84bac7ee25ac0cf8a7159d25f62555625"`
-  and what OCI defines - digest of manifest

I assume we might need to add two things at once (`oci.image.digest` and `?image?.id`). Also `image` namespace would be too generic and cause conflicts with images that are pictures or VM images that are not necessarily in containers.

I'm totally open to having `container.image.id` to be a vague thing specific to runtime (we can explicitly mention what it is in docker and k8s) and then defining something else for OCI standard fields in `oci` or another namespace separately if we're not ready now

Reply:
I add some documentation regarding docker container image id specifics in this commit 5d80f9f4c5ad54aefaabdd01d1e97b51dba34035 in order to unblock this. 

Let me know your thoughts.

Reply:
What do you mean by `here`? I added the docker+container id format as an example in the table and specific docker clarification here https://github.com/open-telemetry/semantic-conventions/pull/39/commits/da791f2993b12effe95122f3aae7a94a9c519e09#diff-505f1d3262fe3c9783be18b6754e7d2c30793d09e9618fbe92d19f94def4fcb8R22 or do you mean something different?

Reply:
Maybe it's related to having docker example + the k8s example @lmolkova mentioned above 

> what k8s uses - a link to the container registry repo with digest "imageID": "registry.azurecr.io/namespace/service/dockerfile@sha256:bdeabd40c3a8a492eaf9e8e44d0ebbb84bac7ee25ac0cf8a7159d25f62555625"
?

Reply:
Done in ca4d5fb7032b1e702a7268558b82148ba4846192.

Reply:
Digging more into this I realize that imageID and imageDigest are different things in docker, k8s and OCI in general. I tried to summarize my view at https://github.com/open-telemetry/semantic-conventions/issues/48#issuecomment-1600350858. So I think it's ok to go for the `container.image.id` here and then cover the digest part in a different field called `container.image.hash` or `container.image.digest` which should be array of strings.

Reply:
@ChrsMark  Sounds good to me :)

Reply:
Yes, I'd definitely address it outside the scope of this PR :)